share/tools/web_config/js/colorutils.js

```diff
@@ -81,8 +81,8 @@ function adjust_lightness(color_str, func) {
     }
 
     /* More hacks */
-    if (color_str == 'black') color_str = '000000';
-    if (color_str == 'white') color_str = 'c0c0c0';
+    if (color_str == 'black') color_str = '#000000';
+    if (color_str == 'white') color_str = '#c0c0c0';
```

Comment:
Nice find. But won't this break the following line (i.e. return NaN)?

```
rgb = parseInt(color_str, 16)
```

Also, the beginning of this function explicitly strips `#` from inputs so adding it back in seems confusing at best. I wonder if there's a simpler way to fix the color issue you've found.

Reply:
Thanks for calling that out... errant change from a find+replace over the file that I missed 😞 .

I haven't found a _better_ way yet to fix the issue, at least in my opinion.  Another thought that I first attempted was to apply a function in the angular attributes to apply a function to insert the `#` as needed.  This felt more hacky and non-obvious maintenance wise to me.

Interesting to note that this bug does _not_ manifest in chrome, but does in firefox.  Peeked a bit at the firefox source and it's hard to dig up where these conversions may take place with only a quick glance.  Firefox uses `rust-cssparser` for some color parsing (maybe just pure CSS and not HTML attributes?) and that project's tests fail if given a hex color without the leading `#`.  There may also be things going on in the angularjs code, though nothing I've found thus far.

Reply:
> Another thought that I first attempted was to apply a function in the angular attributes to apply a function to insert the # as needed. This felt more hacky and non-obvious maintenance wise to me.

I guess that's what I did in https://github.com/charego/fish-shell/commit/b0bb1d0917a9f89b90d8df57b39a25500fea0a03. It does feel a bit hacky. However, I like that the theme code defines the colors in "fish color" format (without leading '#'), and there is a translation layer for display in the browser.

> Interesting to note that this bug does not manifest in chrome, but does in firefox.

Confirmed this. I wonder if the ancient AngularJs code can be blamed. The [color picker](https://github.com/fish-shell/fish-shell/blob/master/share/tools/web_config/partials/colors.html#L110-L119) uses a nested loop and in Firefox their dark entries are missing the "style" attribute. That may be due to the Firefox CSS parser as you mention, but then it's mystifying why it works in most other cases.

```html
<table>
	<tbody>
		<tr data-ng-repeat="color_array in colorArraysArray">
			<td ng-repeat="color in color_array"
			    data-ng-style="{'background-color': color, 'color': color}">...</td>
		</tr>
	</tbody>
</table>
```